source/request_response.c

```diff
@@ -483,9 +487,19 @@ int aws_http_message_get_request_method(
     AWS_PRECONDITION(out_method);
     AWS_PRECONDITION(request_message->request_data);
 
-    if (request_message->request_data && request_message->request_data->method) {
-        *out_method = aws_byte_cursor_from_string(request_message->request_data->method);
-        return AWS_OP_SUCCESS;
+    if (request_message->request_data) {
+        switch (request_message->http_version) {
+            case AWS_HTTP_VERSION_1_1:
+                if (request_message->request_data->method) {
+                    *out_method = aws_byte_cursor_from_string(request_message->request_data->method);
+                    return AWS_OP_SUCCESS;
+                }
+                break;
+            case AWS_HTTP_VERSION_2:
+                return aws_h2_headers_get_request_method(request_message->headers, out_method);
+            default:
+                return aws_raise_error(AWS_ERROR_UNIMPLEMENTED);
```

Comment:
trivial: if this fails due to NOT_AVAILABLE we zero the out-param, but we're not zeroing it here

be consistent